services/ymax-planner/test/kv-store-flow.test.ts

```diff
@@ -0,0 +1,214 @@
+
+  const currentTimeMs = 1700000000; // 2023-11-14T22:13:20Z
+  const txTimestampMs = currentTimeMs - 10 * 1000; // 10 seconds ago
+  const avgBlockTimeMs = 300; // 300 ms per block on eip155:42161
+  mockProvider.getBlockNumber = async () => latestBlock;
+
+  mockProvider.getBlock = async (blockNumber: number) => {
+    const blocksAgo = latestBlock - blockNumber;
+    const ts = currentTimeMs - blocksAgo * avgBlockTimeMs;
+    return { timestamp: Math.floor(ts / 1000) };
+  };
+
+  // Trigger block event to resolve waitForBlock
+  setTimeout(() => mockProvider.emit('block', latestBlock + 1), 10);
+  const event = createMockGmpExecutionEvent(txId);
+
+  mockProvider.getLogs = async (args: {
+    fromBlock: number;
+    toBlock: number;
+  }) => {
+    if (blockWithEvent >= args.fromBlock && blockWithEvent <= args.toBlock)
+      return [event];
+    else return [];
+  };
+
+  const ctxWithFetch = harden({
+    ...opts,
+    fetch: async (url: string) => {
+      return {
+        ok: true,
+        json: async () => ({
+          data: [
+            {
+              status: 'executed',
+              call: {
+                transactionHash: '0xabcdef123456',
+                returnValues: {
+                  messageId: `msg_${txId}`,
+                },
+              },
+              executed: {
+                transactionHash: '0xexecuted123',
+                receipt: {
+                  logs: [createMockGmpExecutionEvent(txId)],
+                },
+              },
+            },
+          ],
+        }),
+      } as Response;
+    },
+  });
+
+  return {
+    ctxWithFetch,
+    gmpTx,
+    mockLog,
+    logs,
+    txTimestampMs,
+  };
+};
+
+test('updates KV store when a pending tx is being searched', async t => {
+  const latestBlock = 100;
+  const txId = 'tx2';
+  const contractAddress = '0x8Cb4b25E77844fC0632aCa14f1f9B23bdd654EbF';
+
+  const { ctxWithFetch, gmpTx, mockLog, txTimestampMs } = setupEnvironment({
+    contractAddress,
+    latestBlock,
+    blockWithEvent: 99,
+    txId,
+  });
+
+  const initialMulticallExecutedSavedBlock = await getTxBlockLowerBound(
+    ctxWithFetch.kvStore,
+    txId,
+    EVENTS.MULTICALL_EXECUTED,
+  );
+  const initialMulticallStatusSavedBlock = await getTxBlockLowerBound(
+    ctxWithFetch.kvStore,
+    txId,
+    EVENTS.MULTICALL_STATUS,
+  );
+
+  t.is(initialMulticallExecutedSavedBlock, undefined);
+  t.is(initialMulticallStatusSavedBlock, undefined);
+
+  await handlePendingTx(
+    { txId, ...gmpTx },
+    {
+      ...ctxWithFetch,
+      log: mockLog,
+    },
+    txTimestampMs,
+  );
+
+  const finalMulticallExecutedSavedBlock = await getTxBlockLowerBound(
+    ctxWithFetch.kvStore,
+    txId,
+    EVENTS.MULTICALL_EXECUTED,
+  );
+  const finalMulticallStatusSavedBlock = await getTxBlockLowerBound(
+    ctxWithFetch.kvStore,
+    txId,
+    EVENTS.MULTICALL_STATUS,
+  );
+
+  t.is(finalMulticallExecutedSavedBlock, 89);
+  t.is(finalMulticallStatusSavedBlock, 89);
```

Comment:
Can you explain why we should expect both of these to be 89 after `handlePendingTx`?